src/course-home/mentoring-tab/ChatWindow.jsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useRef, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { getConfig } from '@edx/frontend-platform';
+import {
+  Avatar, Card, Form, StatefulButton, Stack, Alert,
+} from '@openedx/paragon';
+
+import { CHAT_STATUSES_MAP } from './constants';
+import { useWebSocket } from './hooks';
+import messages from './messages';
+
+const ChatWindow = () => {
+  const [input, setInput] = useState('');
+  const [thinking, setThinking] = useState(false);
+  const [chatMessages, setChatMessages] = useState([]);
+  const [status, setStatus] = useState(CHAT_STATUSES_MAP.default);
+  const lastMessageRef = useRef(null);
+
+  const intl = useIntl();
+  const { OPENEDX_AI_SOCKET_DOMAIN } = getConfig();
+  const disabledStates = [CHAT_STATUSES_MAP.pending, CHAT_STATUSES_MAP.error];
+  const isNotAllowedToSend = !input.trim() || disabledStates.includes(status);
+  const { courseId } = useSelector(state => state.courseHome);
+
+  const { sendMessage } = useWebSocket({
+    url: `ws://${OPENEDX_AI_SOCKET_DOMAIN}/ws/chatgpt/${courseId}/`,
```

Comment:
[question]: Have you tried to test this functionality on different courses?

Reply:
It would also be useful to understand what **limitations** we have related to the course size + how the provision of course context works if it is empty.